eclipse/che
# Bitbucket authorizing using personal access token

### What does this PR do?
Allows Bitbucket authorizing factories using personal access token

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18386

### How to test this PR?
1) Prepare Bitbucket instance with private repository and create personal user token;
2) Prepare namespace for Che user (typically, name should be `<username>-che`);
3) In user-s namespace, create secret containing personal token as follows (change example values to yours):
```
apiVersion: v1
kind: Secret
metadata:
  name: bitbucket-personal-access-token-secret
  labels:
    app.kubernetes.io/part-of: che.eclipse.org
    app.kubernetes.io/component: scm-personal-access-token
  annotations:
    che.eclipse.org/expired-after: '-1'
    che.eclipse.org/che-userid: '355d1ce5-990e-.... '  
    che.eclipse.org/scm-userid: '2'
    che.eclipse.org/scm-username: 'foo'
    che.eclipse.org/scm-url: 'https://bitbucket-bitbucket.apps.cluster-max-ccc6.max-ccc6.example.opentlc.com'
data:
  token:  <BASE-64 encoded token here>
```
4) Execute factory with that rivate repo URL.

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
